tools/prow_spider.py

```diff
@@ -0,0 +1,62 @@
+#coding:utf-8
+import requests
+from fake_useragent import UserAgent
+import time
+import random
+from lxml import etree
+
+
+class ProwSpider(object):
```

Comment:
@jianzhangbjz 
suggest to use github api. it is stable and more information than html.
(because it is public repo, no need to auth it.)

the cli command is `curl -L   -H "Accept: application/vnd.github+json"    https://api.github.com/repos/openshift/release/contents/ci-operator/config/openshift/openshift-tests-private/?ref=master` for example.

of course, we could use python request to get it and then parse json file.

Reply:
if we have one file name, for example xxx.yaml,
it is `curl -L   -H "Accept: application/vnd.github+json"    https://api.github.com/repos/openshift/release/contents/ci-operator/config/openshift/openshift-tests-private/xxx.yaml?ref=master`

it could also reduce request frequency

Reply:
Thanks for the suggestion. Yes, it's better to use the `Github API`, but it also has rate limits as we did before if no token, anyway, I will use the Github API and token for it. 
```yaml
{'message': "API rate limit exceeded for 36.143.64.98. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)", 'documentation_url': 'https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'}
```